src/services/user/user-router.test.ts

```diff
@@ -21,41 +149,100 @@ describe("GET /", () => {
 
     it("gives an not found error for an non-existent user", async () => {
         await Models.UserInfo.deleteOne({
-            userId: TESTER.id,
+            userId: TESTER_USER.userId,
         });
 
-        const response = await getAsAttendee("/user/").expect(400);
+        const response = await getAsAttendee("/user/").expect(404);
 
         expect(JSON.parse(response.text)).toHaveProperty("error", "UserNotFound");
     });
 
     it("works for an attendee user", async () => {
         const response = await getAsAttendee("/user/").expect(200);
 
-        expect(JSON.parse(response.text)).toMatchObject({
-            userId: TESTER.id,
-            name: TESTER.name,
-            email: TESTER.email,
-        });
+        expect(JSON.parse(response.text)).toMatchObject(TESTER_USER);
     });
 
     it("works for an staff user", async () => {
         const response = await getAsStaff("/user/").expect(200);
 
-        expect(JSON.parse(response.text)).toMatchObject({
-            userId: TESTER.id,
-            name: TESTER.name,
-            email: TESTER.email,
-        });
+        expect(JSON.parse(response.text)).toMatchObject(TESTER_USER);
     });
 
     it("works for an admin user", async () => {
         const response = await getAsAdmin("/user/").expect(200);
 
-        expect(JSON.parse(response.text)).toMatchObject({
-            userId: TESTER.id,
-            name: TESTER.name,
-            email: TESTER.email,
+        expect(JSON.parse(response.text)).toMatchObject(TESTER_USER);
+    });
+});
+
+describe("GET /:USERID/", () => {
+    it("gives an forbidden error for a non-staff user", async () => {
+        const response = await getAsAttendee(`/user/${OTHER_USER.userId}/`).expect(403);
+
+        expect(JSON.parse(response.text)).toHaveProperty("error", "Forbidden");
+    });
+
+    it("gives an not found error for a non-existent user", async () => {
+        await Models.UserInfo.deleteOne({
+            userId: OTHER_USER.userId,
+        });
+
+        const response = await getAsStaff(`/user/${OTHER_USER.userId}/`).expect(404);
+
+        expect(JSON.parse(response.text)).toHaveProperty("error", "UserNotFound");
+    });
+
+    it("works for a non-staff user requesting themselves", async () => {
+        const response = await getAsAttendee(`/user/${TESTER_USER.userId}/`).expect(200);
+
+        expect(JSON.parse(response.text)).toMatchObject(TESTER_USER);
+    });
+
+    it("works for a staff user", async () => {
+        const response = await getAsStaff(`/user/${OTHER_USER.userId}/`).expect(200);
+
+        expect(JSON.parse(response.text)).toMatchObject(OTHER_USER);
+    });
+});
+
+describe("POST /", () => {
+    it("gives an unauthorized error for an non-staff user", async () => {
+        const response = await postAsAttendee("/user/").send(JSON.stringify(TESTER_USER_WITH_NEW_EMAIL)).expect(403);
+
```

Comment:
Why does this throw? If we're a user, we should be able to create via auth, no?

Reply:
Removed POST